.eslintrc.js

```diff
@@ -0,0 +1,263 @@
+      "plugin:@typescript-eslint/recommended",
+      "plugin:@typescript-eslint/recommended-requiring-type-checking",
+      "plugin:react/recommended",
+      "plugin:react-hooks/recommended",
+    ],
+    files: ["resources/assets/lib/**/*.{js,jsx,ts,tsx}"],
+    parser: "@typescript-eslint/parser",
+    parserOptions: {
+      project: "tsconfig.json",
+      sourceType: "module",
+    },
+    plugins: [
+      "eslint-plugin-react",
+      "@typescript-eslint",
+    ],
+    rules: {
+      "@typescript-eslint/array-type": [
+        "error",
+        {
+          default: "array",
+        },
+      ],
+      "@typescript-eslint/consistent-type-assertions": "error",
+      "@typescript-eslint/consistent-type-definitions": "error",
+      "@typescript-eslint/dot-notation": "error",
+      "@typescript-eslint/explicit-member-accessibility": [
+        "error",
+        {
+          accessibility: "no-public",
+        },
+      ],
+      "@typescript-eslint/explicit-module-boundary-types": "off",
+      "@typescript-eslint/indent": [
+        "error",
+        2,
+        {
+          FunctionDeclaration: {
+            parameters: "first",
+          },
+          FunctionExpression: {
+            parameters: "first",
+          },
+          SwitchCase: 1,
+        },
+      ],
+      "@typescript-eslint/member-delimiter-style": [
+        "error",
+        {
+          multiline: {
+            delimiter: "semi",
+            requireLast: true,
+          },
+          singleline: {
+            delimiter: "semi",
+            requireLast: false,
+          },
+        },
+      ],
+      "@typescript-eslint/member-ordering": [
+        "error",
+        {
+          default: [
+            "public-static-field",
+            "protected-static-field",
+            "private-static-field",
+
+            "public-instance-field",
+            "protected-instance-field",
+            "private-instance-field",
+
+            "public-constructor",
+            "protected-constructor",
+            "private-constructor",
+
+            "public-static-method",
+            "protected-static-method",
+            "private-static-method",
+
+            "public-instance-method",
+            "protected-instance-method",
+            "private-instance-method",
+          ],
+        },
+      ],
+      "@typescript-eslint/naming-convention": "off",
+      "@typescript-eslint/no-explicit-any": "off",
+      "@typescript-eslint/no-parameter-properties": "off",
+      "@typescript-eslint/no-unsafe-assignment": "warn",
+      "@typescript-eslint/no-unsafe-call": "warn",
+      "@typescript-eslint/no-unsafe-member-access": "warn",
+      "@typescript-eslint/no-unsafe-return": "warn",
+      "@typescript-eslint/no-unused-expressions": "error",
+      "@typescript-eslint/no-use-before-define": "off",
+      "@typescript-eslint/prefer-for-of": "error",
+      "@typescript-eslint/prefer-function-type": "error",
+      "@typescript-eslint/quotes": [
```

Comment:
should be in global for non-ts as well (the non-ts version). And a few other rules as well. [This may be useful](https://github.com/standard/eslint-config-standard-with-typescript/blob/a74134d087afb8eb2707635b85558b31226d2dd3/src/index.ts#L4).